datasetinsights/io/downloader/base.py

```diff
@@ -19,15 +19,16 @@ def _find_downloader(source_uri):
     """
     protocols = "|".join(_registry.keys())
     pattern = re.compile(f"({protocols})")
+
     protocol = pattern.findall(source_uri)
-    if protocol:
+
+    if source_uri.startswith(("https://", "http://")):
```

Comment:
If protocol can't be matched, this will fail. `None string have no method called start_with...`. This should be fixed and add unittest tests for such cases.

Reply:
Are you sure. its the source_uri that is being called not protocol

Reply:
You are correct. 👍